framework/keyring-interface.md

```diff
@@ -75,66 +86,65 @@ perform the [Encrypt Data Key](#encrypt-data-key) behavior.
 
 #### Encrypt Data Key
 
-If the [encryption materials](#structures.md#encryption-materials) contain a plaintext data key,
-OnEncrypt MAY encrypt a data key.
-If the encryption materials do not contain a plaintext data key, OnEncrypt MUST NOT encrypt a data key.
+If the input contains a plaintext data key, or if the keyring generates a data key,
+OnEncrypt MAY encrypt the data key.
+Otherwise, OnEncrypt MUST NOT encrypt a data key.
 
-Encrypt Data Key MAY modify the following fields in the [encryption materials](#structures.md#encryption-materials):
+Encrypt Data Key MAY populate the following fields in the [data key materials](#structures.md#data-key-materials):
 
 - [encrypted data keys](#structures.md#encrypted-data-keys)
 - [keyring trace](#structures.md#keyring-trace)
 
 To perform this behavior, the keyring creates one or more [encrypted data keys](#structures.md#encrypted-data-key)
-using the plaintext data key from the [encryption materials](#structures.md#encryption-materials) as input,
-and appends the [encrypted data keys](#structures.md#encrypted-data-key) to the encrypted data key list
-in the [encryption materials](#structures.md#encryption-materials).
+using the plaintext data key as input,
+and includes the [encrypted data keys](#structures.md#encrypted-data-key)
+in the [data key materials](#structures.md#data-key-materials) result.
 
 The [encrypted data keys](#structures.md#encrypted-data-key) produced by this keyring MUST
 have [ciphertexts](#structures.md#ciphertext) that can be decrypted to the plaintext data key in the
-[encryption materials](#structures.md#encryption-materials).
+[data key materials](#structures.md#data-key-materials).
 
-If OnEncrypt updates the [encryption materials](#data-structure.md#encryption-materials) with at least
-one new [encrypted data key](#structures.md#encrypted-data-key),
+If OnEncrypt outputs [data key materials](#structures.md#data-key-materials) with at least
+one [encrypted data key](#structures.md#encrypted-data-key),
 the [keyring trace](#structures.md#keyring-trace) returned by OnEncrypt MUST include at least one trace
```

Comment:
This wording changes the intent, back to my point about return values and data key material composition.